src/cargo/core/resolver/mod.rs

```diff
@@ -789,12 +775,27 @@ impl RemainingCandidates {
                 }
             }
 
+            // Otherwise the `links` key in the manifest dictates that there's only one
```

Comment:
Noticed the code moved. Any specific reason?

Reply:
Conceptually I think  maintaining "one per  compatibility range"  is more fundamental to how cargo is resolver works then the links attributes, "Otherwise" really belongs on the links. Also, I suspect that we hit the semver  check far more often than the links check, but it was not measurable in benchmarking. Branch predictors are really good.

Reply:
Thanks for the explanation. I think I have guessed your reason behind correctly :)